osutil/disks/disks.go

```diff
@@ -173,6 +195,89 @@ func diskFromMountPointImpl(mountpoint string, opts *Options) (*disk, error) {
 		return nil, fmt.Errorf("cannot find disk for partition %s: %v", partMountPointSource, err)
 	}
 
+	if opts != nil && opts.IsDecryptedDevice {
+		// verify that the mount point is indeed a mapper device, it should:
+		// 1. have DEVTYPE == disk from udev
+		// 2. have dm files in the sysfs entry for the maj:min of the device
+		if props["DEVTYPE"] != "disk" {
+			// not a decrypted device
+			return nil, fmt.Errorf("mountpoint source %s is not a decrypted device: devtype is not disk (is %s)", partMountPointSource, props["DEVTYPE"])
+		}
+
+		// TODO:UC20: currently, we effectively parse the DM_UUID env variable
+		//            that is set for the mapper device volume, but doing so is
+		//            actually wrong, since the value of DM_UUID is an
+		//            implementation detail that depends on the subsystem
+		//            "owner" of the device such that the prefix is considered
+		//            the owner and the suffix is private data owned by the
+		//            subsystem. In our case, in UC20 initramfs, we have the
+		//            device "owned" by systemd-cryptsetup, so we should ideally
+		//            parse that the first part of DM_UUID matches CRYPT- and
+		//            then use `cryptsetup status` (since CRYPT indicates it is
+		//            "owned" by cryptsetup) to get more information on the
+		//            device sufficient for our purposes to find the encrypted
+		//            device/partition underneath the mapper.
+		//            However we don't currently have cryptsetup in the initrd,
+		//            so we can't do that yet :-(
+
+		// TODO:UC20: these files are also likely readable through udev env
+		//            properties, but it's unclear if reading there is reliable
+		//            or not, given that these variables have been observed to
+		//            be missing from the initrd previously, and are not
+		//            available at all during userspace on UC20 for some reason
+		dmUUID, err := ioutil.ReadFile(filepath.Join(devBlockDir, d.Dev(), "dm", "uuid"))
+		if err != nil && os.IsNotExist(err) {
```

Comment:
It looks like we do not handle err != nil errors that are *not* IsNotExit() errors? Or am I missing something? Also, do we need to check for IsNotExit() or could we just return the err value as part of the fmt.Errorf()?

Reply:
good catch, I think distinguishing IsNotExist makes for slightly clearer messages, but although unlikely we need indeed to also consider other kind of errors